### PR TITLE
support protobuf/timestamp for pulsar-sql

### PR DESCRIPTION
**Pull request by tjiuming**
fix: https://github.com/apache/pulsar/issues/3554

### Motivation
1. Support protobuf native timestamp for pulsar-sql, so that presto could display formatted timestamp.
2. Del conflict method `testMergeDifferentCleanupSubscription ` in `SinkConfigUtilTest`, to make sure build project successfully.

### Modifications
1. add protobuf/timestamp support in `PulsarProtobufNativeRowDecoderFactory`
2. return the `seconds` field of  protobuf/timestamp  in `PulsarProtobufNativeColumnDecoder`
 
### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (don't know)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (don't know)

### Documentation

Check the box below and label this PR (if you have committer privilege).

Need to update docs? 
  
- [x] `no-need-doc` 
  a small feature, seems no need

### CODE REVIEW

**Comment by tjiuming** [comment]
TBR

**Comment by tjiuming** [comment]
there is a question, protobuf/timestamp contains nanos, but presto/timestamp not. so that nanos will be missed in pulsar-sql  when display formatted timestamp.

**Comment by dao-jun** [comment]
> @tjiuming:Thanks for your contribution. For this PR, do we need to update docs? (The [PR template contains info about doc](https://github.com/apache/pulsar/blob/master/.github/PULL_REQUEST_TEMPLATE.md#documentation), which helps others know more about the changes. Can you provide doc-related info in this and future PR descriptions? Thanks)

I‘m not sure, it is a small feature, just display formatted timestamp, maybe not.

**Comment by tjiuming** [comment]
> @tjiuming:Thanks for your contribution. For this PR, do we need to update docs? (The [PR template contains info about doc](https://github.com/apache/pulsar/blob/master/.github/PULL_REQUEST_TEMPLATE.md#documentation), which helps others know more about the changes. Can you provide doc-related info in this and future PR descriptions? Thanks)

I‘m not sure, it is a small feature, just display formatted timestamp, maybe not.

File: pulsar-functions/utils/src/test/java/org/apache/pulsar/functions/utils/SinkConfigUtilsTest.java [hunk 1]
```diff
@@ -390,22 +390,6 @@ public void testMergeRuntimeFlags() {
         );
     }
 
-    @Test
-    public void testMergeDifferentCleanupSubscription() {
```
Comment (codelipenghui) [inline pulsar-functions/utils/src/test/java/org/apache/pulsar/functions/utils/SinkConfigUtilsTest.java:394]: Could you please rebase the master branch? I think the test has already been removed on the master branch.

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeRowDecoderFactory.java [hunk 2]
```diff
@@ -25,23 +25,13 @@
...
-import io.prestosql.spi.type.IntegerType;
-import io.prestosql.spi.type.RealType;
-import io.prestosql.spi.type.RowType;
-import io.prestosql.spi.type.StandardTypes;
-import io.prestosql.spi.type.Type;
-import io.prestosql.spi.type.TypeManager;
-import io.prestosql.spi.type.TypeSignature;
-import io.prestosql.spi.type.TypeSignatureParameter;
-import io.prestosql.spi.type.VarbinaryType;
+import io.prestosql.spi.type.*;
```
Comment (codelipenghui) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeRowDecoderFactory.java:33]: Please avoid use star import.

**Review by codelipenghui — CHANGES_REQUESTED** [review]
Could you please add a test to cover the new fix? So that we can avoid the issue will not happening in the future.

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeColumnDecoder.java [hunk 3]
```diff
@@ -192,6 +194,12 @@ public long getLong() {
                 return floatToIntBits((Float) value);
             }
 
+            //return seconds field of protobuf/timestamp
+            if (columnType instanceof TimestampType && value instanceof DynamicMessage) {
+                DynamicMessage message = (DynamicMessage) value;
+                return (long) message.getField(message.getDescriptorForType().findFieldByName("seconds"));
```
Comment (codelipenghui) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeColumnDecoder.java:200]: What is the difference between the following pattern:

```
            if (columnType instanceof TimestampType && value instanceof Timestamp) {
                return ((Timestamp) value).getSeconds();
            }
```

Comment (tjiuming) [reply pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeColumnDecoder.java:200]: I’ll check it, but I confirmed `value instanceof DynamicMessage` is true, and protobuf/timestamp not extends DynamicMessage, if I use `value instanceof DynamicMessage`, it seems not work

Comment (tjiuming) [reply pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeColumnDecoder.java:200]: I’ll check it, but I confirmed `value instanceof DynamicMessage` is true, and protobuf/timestamp not extends DynamicMessage, if I use `value instanceof Timestamp`, it seems not work

File: pulsar-functions/utils/src/test/java/org/apache/pulsar/functions/utils/SinkConfigUtilsTest.java (hunk 1 above)
Comment (tjiuming) [reply pulsar-functions/utils/src/test/java/org/apache/pulsar/functions/utils/SinkConfigUtilsTest.java:394]: the method still exists yesterday, but I checkout just now, it removed.

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeRowDecoderFactory.java (hunk 2 above)
Comment (tjiuming) [reply pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeRowDecoderFactory.java:33]: ok, got it

**Comment by tjiuming** [comment]
@codelipenghui  TBR

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeColumnDecoder.java (hunk 3 above)
Comment (RobertIndie) [reply pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeColumnDecoder.java:200]: I think we should return time in the unit of milliseconds here.  It seems that we need to convert it here. And do we also need to handle `Timestamp. nanos` here?

File: pulsar-sql/presto-pulsar/src/test/java/org/apache/pulsar/sql/presto/decoder/protobufnative/TestProtobufNativeDecoder.java [hunk 4]
```diff
@@ -83,6 +85,7 @@ public void testPrimitiveType() {
                 .setBoolField(true)
                 .setBytesField(ByteString.copyFrom("abc".getBytes()))
                 .setTestEnum(TestMsg.TestEnum.FAILOVER)
+                .setTimestampField(Timestamp.newBuilder().setSeconds(System.currentTimeMillis()).build())
```
Comment (RobertIndie) [inline pulsar-sql/presto-pulsar/src/test/java/org/apache/pulsar/sql/presto/decoder/protobufnative/TestProtobufNativeDecoder.java:88]: We should use the time in seconds here.

**Review by RobertIndie — COMMENTED** [review]
I have tested it in C# client. But the time shown in pulsar SQL is wrong, see `lastlogintimestamp`:
```
presto> select * from pulsar."public/default".test;
 name | age |   lastlogintimestamp    | __partition__ |     __event_time__      |    __publish_time__     | __message_id__ | __sequence_id__ | __producer_name__ | __key__ | __properties__
------+-----+-------------------------+---------------+-------------------------+-------------------------+----------------+-----------------+-------------------+---------+----------------
 abc  |  18 | 1970-01-20 07:39:05.115 |            -1 | 1970-01-01 08:00:00.000 | 2021-12-24 11:25:15.093 | (76,0,0)       |               0 | NULL              | NULL    | {}
```
The correct time is `2021-12-24 ... ...`.

**Review by RobertIndie — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**